File: Back-End/src/main/java/com/controle/combustivel/domain/model/SaidaCombustivel.java
package com.controle.combustivel.domain.model;

import java.io.Serializable;
import java.time.OffsetDateTime;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import lombok.Data;

@Entity
@Data
public class SaidaCombustivel {
	
	private static final long serialVersionUID = 1L;
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private long id;
	@ManyToOne
	@JoinColumn(nullable = false)
	private Reservatorio origem;
	@ManyToOne
	@JoinColumn(nullable = false)
	private Maquina destino;
	@ManyToOne
	@JoinColumn(nullable = false)
	private Abastecedor abastecedor;
	private double quantidade;
	@ManyToOne
	@JoinColumn(nullable = false)
	private Combustivel combustivel;
	@Column(nullable = false, columnDefinition = "dateTime")
	private OffsetDateTime dataHoraSaida;

	
	
}

File: ColetorJetPackCompose/app/src/main/java/com/example/coletorjetpackcompose/model/SaidaCombustivel.kt
package com.example.coletorjetpackcompose.model

import java.time.OffsetDateTime

data class SaidaCombustivel(

    var origem: String? = null,


    var destino: String? = null,


    var abastecedor: Int? = null,

    var quantidade: Int? = null,

    var combustivel: Int? = null,
    var combustivelText: String? = null,
    var isEnviadoVisible : Boolean = true,


    var dataHoraSaida: String? = null
)
